src/multinode_run.jl

```diff
@@ -1,193 +1,700 @@
+                if curr_cbox > NumConcurrentBoxes
+                    curr_cbox = 1
+                end
+                cbox = conc_boxes[curr_cbox]
+                if cbox.state[] != BoxLoaded
+                    if !load_box(cbox, mi, all_boxes, stagedir;
+                                 primary_initialization=primary_initialization,
+                                 timing=times)
+                        break
+                    end
+                end
+            end
         end
+    end
 
-        box = all_boxes[box_idx]
+    if nthreads() == 1
+        process_boxes()
+    else
+        ccall(:jl_threading_run, Void, (Any,), Core.svec(process_boxes))
+        #ccall(:jl_threading_profile, Void, ())
+    end
+    nputs(grank(), "completed at $(Dates.now())")
+
+    for i = 1:nthreads()
+        add_timing!(timing, ttimes[i])
+    end
+
+    tic()
+    save_results(all_results, outdir)
+    timing.write_results = toq()
+
+    tic()
+    finalize(mi.dt)
+    timing.wait_done = toq()
+    nputs(grank(), "synchronized ranks at $(Dates.now())")
+end
 
-        ntputs(nodeid, threadid(), "loading box $(box.ramin), $(box.ramax), $(box.decmin), $(box.decmax)")
 
-        # load this box
-        tic()
-        rcfs = get_overlapping_fields(box, stagedir)
-        timing.query_fids = timing.query_fids + toq()
+"""
+Return a Cyclades partitioning or an equal partitioning of the target sources.
+"""
+function partition_box(npartitions::Int, target_sources::Vector{Int},
+                       neighbor_map::Vector{Vector{Int}};
+                       cyclades=true,
+                       batch_size=400)
+    if cyclades
+        cyclades_neighbor_map = Dict{Int64,Vector{Int64}}()
+        for (index, neighbors) in enumerate(neighbor_map)
+            cyclades_neighbor_map[target_sources[index]] = neighbors
+        end
+        return partition_cyclades(npartitions, target_sources, cyclades_neighbor_map,
+                                  batch_size=batch_size)
+    else
+        return partition_equally(npartitions, length(target_sources))
+    end
+end
+
 
-        cbox.catalog, cbox.sources,
-        cbox.neighbor_map, cbox.images = infer_init(rcfs, stagedir;
-                                                    primary_initialization=primary_initialization,
-                                                    timing=timing)
-        cbox.curr_source = 1
-        cbox.sources_done = 0
+"""
+Initialize elbo args for the specified target source.
+"""
+function init_elboargs(ts::Int, catalog::Vector{CatalogEntry},
+                       target_sources::Vector{Int},
+                       neighbor_map::Vector{Vector{Int}}, images::Vector{Image},
+                       ts_vp::Dict{Int64,Array{Float64}},
+                       ea_vec::Vector{ElboArgs},
+                       cfg_vec::Vector{Config{DEFAULT_CHUNK,Float64}};
+                       use_fft=false,
+                       trust_region=NewtonTrustRegion())
```

Comment:
These `use_fft` and `trust_region` flags aren't used any more in `joint_infer.jl`. Can you call the code in `joint_infer` to initialize elbo args, either `initialize_elboargs_sources!` or `initialize_elboargs_sources_kernel!` rather than duplicating the code here?